otter_welcome_buddy/gql_service/models/gql_leetcode_model.py

```diff
@@ -0,0 +1,706 @@
+    reputation: int | None = None
+    reputation_diff: int | None = None
+    solution_count: int | None = None
+    solution_count_diff: int | None = None
+    category_discuss_count: int | None = None
+    category_discuss_count_diff: int | None = None
+
+
+class LeetcodeUserModel(BaseGqlModel):
+    """
+    Data class representing a user on Leetcode.
+
+    Attributes:
+        username (str): The username of the Leetcode user.
+        profile (LeetcodeUserProfileModel): The profile information of the user.
+        contributions (LeetcodeUserContributionModel): The contributions made by the user.
+        github_url (str): The GitHub URL of the user.
+        twitter_url (str): The Twitter URL of the user.
+        linkedin_url (str): The LinkedIn URL of the user.
+        submission_calendar (str): JSON encoded string representing the submission calendar where
+                                        each entry is a timestamp and the number of submissions.
+        submit_stats_global (LeetcodeUserSubmitStatsModel): The global submission statistics of the
+                                                            user.
+        submit_stats (LeetcodeUserSubmitStatsModel): The submission statistics of the user.
+        active_badge (LeetcodeUserBadgeModel): The active badge of the user.
+        contest_badge (LeetcodeUserBadgeModel): The contest badge of the user.
+        tag_problem_counts (LeetcodeTagProblemCountsCategoryModel): The problem counts categorized
+                                                                    by tags.
+        language_problem_count (list[LeetcodeLanguageProblemCountModel]): The problem counts
+                                                        categorized by programming languages.
+        is_discuss_admin (bool): Indicates if the user is a discuss admin.
+        is_discuss_staff (bool): Indicates if the user is a discuss staff member.
+        is_active (bool): Indicates if the user is currently active.
+    """
+
+    username: str | None = None
+    profile: LeetcodeUserProfileModel | None = None
+    contributions: LeetcodeUserContributionModel | None = None
+    github_url: str | None = None
+    twitter_url: str | None = None
+    linkedin_url: str | None = None
+    # Encode JSON object where each entry represent:
+    # {
+    #   timestamp: # submissions
+    # }
+    submission_calendar: str | None = None
+    submit_stats_global: LeetcodeUserSubmitStatsModel | None = None
+    submit_stats: LeetcodeUserSubmitStatsModel | None = None
+    active_badge: LeetcodeUserBadgeModel | None = None
+    contest_badge: LeetcodeUserBadgeModel | None = None
+    tag_problem_counts: LeetcodeTagProblemCountsCategoryModel | None = None
+    language_problem_count: list[LeetcodeLanguageProblemCountModel] | None = None
+    is_discuss_admin: bool | None = None
+    is_discuss_staff: bool | None = None
+    is_active: bool | None = None
+    # nameColor - nullable
```

Comment:
Consider removing the non used coded, or if its is going to be used in the future, comment why it is commented rn.